### PR TITLE
Add support for token exchange using shared links

### PR DESCRIPTION
**Pull request by boxdave**
- Adds shared_link kwarg to Client.downscope_token. If shared link is sent as an arg in the method call, the downscoped token will be scoped to the resource the shared link references.
- Adds unit test for Client.downscope_token with share_link sent as an arg, and shared_link and item not sent as an arg.

### CODE REVIEW

File: test/unit/client/test_client.py [hunk 1]
```diff
@@ -1072,7 +1072,7 @@ def check_downscope_token_request(
         mock_object_id,
         make_mock_box_request,
 ):
-    def do_check(item_class, scopes, additional_data, expected_data):
```
Comment (sujaygarlanka) [inline test/unit/client/test_client.py]: Why did you change the order of these parameters?

File: boxsdk/client/client.py [hunk 2]
```diff
@@ -1399,7 +1398,7 @@ def get_pending_collaborations(self, limit=None, offset=None, fields=None):
         )
 
     @api_call
-    def downscope_token(self, scopes, item=None, additional_data=None):
+    def downscope_token(self, scopes, item=None, shared_link=None, additional_data=None):
```
Comment (sujaygarlanka) [inline boxsdk/client/client.py]: Also, the shared_link parameter should be after additional_data to avoid a breaking change.

**Comment by sujaygarlanka** [comment]
@boxdave I updated the HISTORY.rst with this change. For future fixes in the SDKs, do you mind updating the change log. In all other SDKs other than this one, it is the CHANGELOG.md file. This is something new are trying to adhere to in order to make our releases quicker.

File: test/unit/client/test_client.py (hunk 1 above)
Comment (boxdave) [reply test/unit/client/test_client.py]: I changed the order here to match the signature of Client.downscope_token. Minor thing, but in that method, scopes comes first then the item. I thought it seems a little more consistent to match that signature in do_check

File: test/unit/client/test_client.py [hunk 3]
```diff
@@ -1123,7 +1120,35 @@ def test_downscope_token_sends_downscope_request(
         'scope': expected_scopes,
         'grant_type': 'urn:ietf:params:oauth:grant-type:token-exchange',
     }
-    check_downscope_token_request(item_class, scopes, {}, expected_data)
+    check_downscope_token_request(scopes, item_class, None, None, expected_data)
+
+
+def test_downscope_token_sends_downscope_request_with_no_item_shared_link_or_additional_data(
```
[the remark this answers is not in the document]
Comment (dtrodger) [reply test/unit/client/test_client.py]: @sujaygarlanka It looks like this test is actually handled by the last pytest.mark.parametrize in test_downscope_token_sends_downscope_request already. Should I just remove this redundant test?

[the remark this answers is not in the document]
Comment (sujaygarlanka) [reply test/unit/client/test_client.py]: You're right. I would remove this test then.

[the remark this answers is not in the document]
Comment (dtrodger) [reply test/unit/client/test_client.py]: Just pushed a commit that removes this.

File: boxsdk/client/client.py [hunk 4]
```diff
@@ -15,6 +14,7 @@
 from ..pagination.limit_offset_based_object_collection import LimitOffsetBasedObjectCollection
 from ..pagination.marker_based_object_collection import MarkerBasedObjectCollection
 from ..util.shared_link import get_shared_link_header
+from ..object.item import Item
```
Comment (sujaygarlanka) [inline boxsdk/client/client.py]: Can this be removed? Linter is not passing because this is imported, but not used.
